hermes-console-vue/src/composables/subscription/use-create-subscription/useCreateSubscription.ts

```diff
@@ -0,0 +1,160 @@
+import {
+  createSubscription as doCreateSubscription,
+  searchOwners,
+} from '@/api/hermes-client';
+import { parseFormToRequestBody } from '@/composables/subscription/use-form-subscription/form-mapper';
+import { ref, watch } from 'vue';
+import { storeToRefs } from 'pinia';
+import { useAppConfigStore } from '@/store/app-config/useAppConfigStore';
+import { useFormSubscription } from '@/composables/subscription/use-form-subscription/useFormSubscription';
+import { useGlobalI18n } from '@/i18n';
+import { useNotificationsStore } from '@/store/app-notifications/useAppNotifications';
+import type { CreateSubscriptionFormRequestBody } from '@/api/subscription';
+import type {
+  DataSources,
+  SubscriptionForm,
+} from '@/composables/subscription/use-form-subscription/types';
+import type { Ref } from 'vue';
+import type {
+  UseCreateSubscription,
+  UseCreateSubscriptionErrors,
+} from '@/composables/subscription/use-create-subscription/types';
+
+export function useCreateSubscription(topic: string): UseCreateSubscription {
+  const notificationsStore = useNotificationsStore();
+
+  const errors = ref<UseCreateSubscriptionErrors>({
+    fetchOwnerSources: null,
+    fetchOwners: null,
+  });
+  const { form, validators, dataSources } = useFormSubscription();
+  initializeForm(form, dataSources);
+  watch(
+    () => form.value.deliveryType,
+    () => {
+      form.value.contentType = '';
+    },
+  );
+  const searchTimeout = ref();
+  watch(
```

Comment:
this piece of code is the same as in `editSubscription`, maybe we should have some helper functions for searching the owners?